docs/writing_an_aggregate_builder.md

```diff
@@ -0,0 +1,126 @@
+Most examples (and use-cases) for builders and generation are around reading a
+single file, and outputting another file as a result. For example,
+`json_serializable` emits a `<file>.g.dart` per a `<file>.dart` to encode/decode
+to JSON, and `sass_builder` emits a `<file>.css` per a `<file>.scss`.
+
+However, sometimes you want to output one or more files based on the inputs of
+_many_ files, perhaps even all of them. We call this abstractly, an _aggregate
+builder_, or a builder with many inputs and one (or less) outputs.
+
+> **WARNING**: This pattern could have negative effects on your development
+> cycle and incremental build, as it invalidates frequently (if _any_ of the
+> input files change).
```

Comment:
It might not be clear what an 'input' means since it's different from the normal `inputId`. How about "if _any_ read file changes".

Reply:
Done.